packages/@react-stately/data/test/useAsyncList.test.js

```diff
@@ -15,98 +15,173 @@ import React from 'react';
 import {useAsyncList} from '../src';
 
 const ITEMS = [{id: 1, name: '1'}, {id: 2, name: '2'}];
-describe('useAsyncList', () => {
-  let loadSpy = jest.fn().mockImplementation(() => ({items: ITEMS}));
-  let loadMoreSpy = jest.fn().mockImplementation(({items}) => ({items: [...items, ...ITEMS]}));
-  let sortSpy = jest.fn();
+const ITEMS2 = [{id: 2, name: '1'}, {id: 1, name: '2'}];
+
+function getItems() {
+  return new Promise(resolve => {
+    setTimeout(() => resolve({items: ITEMS, cursor: 3}), 100);
+  });
+}
 
-  afterEach(() => {
-    loadMoreSpy.mockClear();
-    loadSpy.mockClear();
-    sortSpy.mockClear();
+function getItems2() {
+  return new Promise(resolve => {
+    setTimeout(() => resolve({items: ITEMS2, cursor: 0}), 100);
+  });
+}
+
+describe('useAsyncList', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
   });
 
-  it('will call load function on init', async () => {
-    let {result, waitForNextUpdate} = renderHook(() => useAsyncList({load: loadSpy}));
+  it('should call load function on init', async () => {
+    let load = jest.fn().mockImplementation(getItems);
+    let {result, waitForNextUpdate} = renderHook(() => useAsyncList({load}));
+
+    expect(load).toHaveBeenCalledTimes(1);
+    let args = load.mock.calls[0][0];
+    expect(args.items).toEqual([]);
+    expect(args.selectedKeys).toEqual(new Set());
+    
+    expect(result.current.isLoading).toBe(true);
+    expect(result.current.items).toEqual([]);
+
     await act(async () => {
+      jest.runAllTimers();
```

Comment:
if we're using jest.runAllTimers inside an act, we shouldn't need the await/async or await waitForNextUpdate
though it's probably a bit hairy because of the promise https://github.com/testing-library/react-hooks-testing-library/issues/241
I'm not sure the best way to handle this :(

Reply:
given that you control the length of time on the timer, maybe only advance the timer by the expected amount?
runAll is a bit heavy and while useful in integration tests, probably not as useful in hook tests

Reply:
I don't quite follow the downside to using runAllTimers here.